nautobot/docs/installation/region-and-site-data-migration-guide.md

```diff
@@ -0,0 +1,438 @@
+
+```python
+# models.py
+class ExampleModel(OrganizationalModel):
+    # note that site field is gone
+    location = models.ForeignKey(
+        to="dcim.Location",
+        on_delete=models.PROTECT,
+        blank=True,
+        null=True,
+    )
+    name = models.CharField(max_length=20, help_text="The name of this Example.")
+...
+```
+
+After removing the `site` attribute, make the migration file by running `nautobot-server makemigrations [app_name] -n [migration_name]`, for example:
+
+```shell
+nautobot-server makemigrations example_app -n remove_site_field_from_example_model
+```
+
+The migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+
+    dependencies = [
+        ('example_app', '0008_migrate_example_model_data_from_site_to_location'),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name='examplemodel',
+            name='site',
+        ),
+    ]
+```
+
+!!! important
+    Before you apply this migration, you have to add a `run_before` attribute in this migration file to make sure that you remove `site`/`region` fields before `Site` and `Region` models themselves are removed. **Without it, your migration files might be out of order and your app will not start**.
+
+```python
+    # Ensure this migration is run before the migration that removes Region and Site Models
+    run_before = [
+        # TODO we need to change the name when PR #3313 is merged and migrations files are reordered.
+        ("dcim", "0034_remove_region_and_site"),
+    ]
+```
+
+The final migration file might look like this:
+
+```python
+# Generated by Django 3.2.17 on 2023-02-22 17:09
+# 0009_remove_site_field_from_example_model.py
+
+from django.db import migrations
+
+class Migration(migrations.Migration):
+    
+    # Ensure this migration is run before the migration that removes Region and Site Models
+    run_before = [
+        # TODO we need to change the name when PR #3313 is merged and migrations files are reordered.
+        ("dcim", "0034_remove_region_and_site"),
+    ]
+    dependencies = [
+        ("example_app", "0008_migrate_example_model_data_from_site_to_location"),
+    ]
+
+    operations = [
+        migrations.RemoveField(
+            model_name="examplemodel",
+            name="site",
+        ),
+    ]
+```
+
+Apply the migration files by running `nautobot-server migrate [app_name]`, for example:
+
+```shell
+nautobot-server migrate example_app
+```
+
+## Region and Site Related Data Model Migration Guide For New Nautobot App installations in an Existing Nautobot 2.0 Environment
```

Comment:
Is this still a work-in-progress? Why an empty section?

Reply:
I need to get an app that has long-running migrations and that uses `Site` and `Region` models in it. So I can try to install it in our alpha environment to know what exactly to do when installing new apps in the 2.0.0 environment.

Reply:
I didn't want to hold up this PR anymore so I thought I could just merge it and add on to it later.

Reply:
It's fine then, if you wouldn't mind submitting that on the backlog so we don't lose track of it if it isn't already?

Reply:
https://github.com/nautobot/nautobot/issues/3387